byway/src/main/webapp/destinations.js

```diff
@@ -0,0 +1,203 @@
+    // For each place, get the icon, name and location.
+    const bounds = new google.maps.LatLngBounds();
+    places.forEach(place => {
+      if (!place.geometry) {
+        console.log("Returned place contains no geometry");
+        return;
+      }
+      const icon = {
+        url: place.icon,
+        size: new google.maps.Size(71, 71),
+        origin: new google.maps.Point(0, 0),
+        anchor: new google.maps.Point(17, 34),
+        scaledSize: new google.maps.Size(25, 25)
+      };
+      // Create a marker for each place.
+      markers.push(
+        new google.maps.Marker({
+          map,
+          icon,
+          title: place.name,
+          position: place.geometry.location
+        })
+      );
+
+      if (place.geometry.viewport) {
+        // Only geocodes have viewport.
+        bounds.union(place.geometry.viewport);
+      } else {
+        bounds.extend(place.geometry.location);
+      }
+    });
+    map.fitBounds(bounds);
+}
+
+/*
+* fetches start location and destinations from DestinationsServlet and adds to DOM
+*/
+function updateLocations(locationData){
+    document.getElementById('start-location').innerText = "Start Location :" + locationData.start;
+    const container = document.getElementById('destinations-container');
+    container.innerText = "Destinations:";
+    let destinationArray= locationData.destinations;
+    destinationArray.forEach((destination) => {
+      let destinationToAdd = document.createElement('p');
+      destinationToAdd.innerText = destination;
+      container.appendChild(destinationToAdd);
+    }) 
+}
+
+/* 
+* fills Start location Searchbox with previously input
+*/
+function updateStartDestination(locationData){
+    if (locationData.start == null){
+      document.getElementById('start-search-box').value = "";
+    }
+    else{
+      document.getElementById('start-search-box').value = locationData.start;
+    }
+}
+
+/* 
+* fetches data from servlet
+*/
+function fetchDestinations(){
+  let promise= fetch('/api/destinations').then(response => response.json());
+  return promise;
+}
+
+/* 
+* Gets users current location 
+*/
+function getCurrentAddress(){
+  const geocoder = new google.maps.Geocoder();
+  geocoder.geocode({ 'location': userlatlng}, (results, status) => {
+    if (status === "OK") {
+      if (results[0]) {
+          console.log(results[0].formatted_address);
+        document.getElementById("start-search-box").value=results[0].formatted_address;
+      } else {
+          window.alert("No results found");
+        }
+    } else {
+        window.alert("Geocoder failed due to: " + status);
+      }
+  });
+}
+
+/* 
+* add event listener for submit button
+*/
+window.onload = function(){
+  document.getElementById('user-input-form').addEventListener('submit', (event) => {
+    event.preventDefault();
+    const formData = new  FormData(document.getElementById("user-input-form"));
+    fetch('/api/destinations', {method: 'POST', body:formData}).then(()=>
```

Comment:
The response to the POST request should be the current `locationData`, same as the response to `fetchDestinations`, right? So you should be able to skip the use of `fetchDestinations` and just convert the value of this `fetch` result to json, just like in `fetchDestinations`.

Reply:
fixed